### PR TITLE
Add support STM32F769 SoC and the STM32F769I-DISCO board

### PR DESCRIPTION
**Pull request by jinyongiot**
This patch series adds basic supports for the STM32F769 SoC on the STM32F769I-DISCO board.

I have tested it running hello-world, base blinky, base threads,etc.

### CODE REVIEW

**Comment by erwango** [comment]
@jinyongiot , this looks better, thanks. 
But you still need to add a one line comment in commit message.

**Comment by erwango** [comment]
@jinyongiot , if you need to add some changes to your PR, you just need to use following command:
git push -f <remote name (:origin?)> master

File: dts/arm/st/mem.h [hunk 1]
```diff
@@ -91,6 +91,9 @@
 #elif defined(CONFIG_SOC_STM32F746XG)
 #define DT_FLASH_SIZE		__SIZE_K(1024)
 #define DT_SRAM_SIZE		__SIZE_K(320)
+#elif defined(CONFIG_SOC_STM32F769XI)
+#define DT_FLASH_SIZE		__SIZE_K(1024)
```
Comment (ydamigos) [inline dts/arm/st/mem.h]: ST site mentions 2048 flash size for STM32F769XI SoCs.

File: dts/arm/st/stm32f769.dtsi [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c)  2018 Yong Jin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <st/stm32f746.dtsi>
+
+
```
Comment (ydamigos) [inline dts/arm/st/stm32f769.dtsi]: Remove the 2 empty lines.

File: boards/arm/stm32f769i_disco/Kconfig.board [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Kconfig - STM32F746G Discovery board configuration
```
Comment (ydamigos) [inline boards/arm/stm32f769i_disco/Kconfig.board]: STM32F769I?

File: boards/arm/stm32f769i_disco/Kconfig.defconfig [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Kconfig - STM32F746G DISCOVERY board configuration
```
Comment (ydamigos) [inline boards/arm/stm32f769i_disco/Kconfig.defconfig]: STM32F769I?

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: dts/arm/st/mem.h (hunk 1 above)
Comment (jinyongiot) [reply dts/arm/st/mem.h]: so sorry!

**Review by erwango — CHANGES_REQUESTED** [review]
One additional comment on top of @ydamigos ones.
Otherwise it's good, thanks!

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	leds {
+		compatible = "gpio-leds";
+		red_led_1: led@0 {
+			gpios = <&gpioj 13 GPIO_INT_ACTIVE_HIGH>;
+			label = "User LD1";
+		};
+		green_led_2:led@1 {
+			gpios = <&gpioj 5 GPIO_INT_ACTIVE_HIGH>;
+			label = "User LD2";
+		};
```
[the remark this answers is not in the document]
Comment (jinyongiot) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts]: I am so sorry for my bad job!
thank you ! but LD4 is PD4

[the remark this answers is not in the document]
Comment (erwango) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts]: @jinyongiot 
> I am so sorry for my bad job!

This is not at all a bad job.
Code review is not only done to point errors, but also to mention that community expectations could be different from your initial target when coding. And we're here to help you submitting a code that will be fully in line with community expectations.
You're doing good for your first contribution, comments do not mean quality is poor.

[the remark this answers is not in the document]
Comment (jinyongiot) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts]: thanks

**Review by dbkinder — APPROVED** [review]
doc LGTM, thanks!

File: boards/arm/stm32f769i_disco/Kconfig.defconfig (hunk 4 above)
Comment (jinyongiot) [reply boards/arm/stm32f769i_disco/Kconfig.defconfig]: Thank you!
This mitake removed now

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jinyongiot) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts]: Now, the LD3 & LD4 added, and some mistakes removed!
Thanks!

**Review by erwango — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Pull request merged**